src/tiled/tileseteditor.cpp

```diff
@@ -196,7 +196,13 @@ void TilesetEditor::addDocument(Document *document)
 
     TilesetView *view = new TilesetView(mWidgetStack);
     view->setTilesetDocument(tilesetDocument);
-    view->setZoomable(new Zoomable(this));
+
+    auto zoomable = new Zoomable(this);
+    auto fileName = tilesetDocument->fileName();
+    qreal zoom = Preferences::instance()->TilesetScale(fileName);
```

Comment:
The Tilesets dock currently displays the zoom level of its own single Zoomable instance, which it sets on each created TilesetView. That way, all tileset views are at the same zoom level. However, you are now storing the zoom level for each tileset, and the Tilesets dock should probably use this per-tileset zoom level as well.

On the other hand, it now occurred to me that you may want to zoom out the tileset in the dock and not in the editor, so I'm now wondering whether it would then make sense to store two zoom levels for each tileset...